### PR TITLE
2022 update

### LINKED ISSUE #26: Talon FX Support
Similar if not exactly the same as the spark max. But it has a higher tick count per rotation. 

Its used for the falcon 500. Uses integrated encoder and integrated motor controller (Talon FX) powered over PWM & CAN.

Curtin is using PWM. & maybe CAN. Probably try to make support for both.

### LINKED ISSUE #45: Gradle Library
Make WML a pre-compiled library using gradle (Possibly maven without needing to add wml as a submodule)

Testing over in branch gradle-lib

### PR DESCRIPTION
**Pull request by CJBuchel**
- Separates WML into WML-Core and WML-Rev.
- Adds NativeLibrarySpec for including as static/shared library into projects
- Adds build environment for azure and testing

closes #45 
closes #26

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,80 +2,71 @@
...
-
-Or read the source code of the library; it has a few comments inside there. But uhhhhh, I don't recommend it. Even I don't know what half the stuff is...
+- Added to the library is an [example folder](/examples/). More examples are planned, however, I am aware that there are not many examples currently (especially compared to the breadth of the content in the library).
 
-That being said, if you have an issue or question feel free to contact us. We don't have lives... we're programmers.
+- I recommend looking at code that uses the library to get a better idea of what to do. (e.g. The Build environment we use for testing [Src](src/main))
 
 
-<sub><sup>readme written by [@CJBuchel](https://github.com/CJBuchel), 4/03/20</sup></sub>
+<sub><sup>readme written by [@CJBuchel](https://github.com/CJBuchel), 27/09/21</sup></sub>
```
Comment (met4000) [inline README.md]: *27/10/21

File: .gitignore [hunk 2]
```diff
@@ -160,3 +160,6 @@ bin/
 # End of https://www.gitignore.io/api/c++,java,linux,macos,gradle,windows,visualstudiocode
 
 !libs/**/*.*
+
+# Ignore Gradle build output directory
+build
```
Comment (met4000) [inline .gitignore]: would recommend clarifying with this as `build/`

**Review by met4000 — DISMISSED** [review]

**Review by CJBuchel — COMMENTED** [review]
Fixed in latest push

**Review by met4000 — APPROVED** [review]
Probably good enough; I absolve from any responsibility of broken code made by this though :)

**Comment by CJBuchel** [comment]
I declare responsibility for broken code.... ish.
Except for NTUtil and the desktop support. Blame nathan

**Pull request merged**